### PR TITLE
Ensure that shoot kube-apiserver has no incompatible mount of usr/share/ca-certificates

### PR DESCRIPTION
**Pull request by danielfoehrKn**
**How to categorize this PR?**

/area control-plane
/kind bug
/priority normal
/platform openstack

**What this PR does / why we need it**:

Previous openstack versions  ensure a volume mount on the shoot kube-apiserver with name `usr-share-ca-certificates` with path `/usr/share/ca-certificates for Shoots that does not use CSI or is not yet migrated. 

However Gardener version > 1.10.0 always adds a volume mount for Shoots with Kubernetes version > 1.17 with name `usr-share-cacerts` also with path `/usr/share/ca-certificates`.

This is problematic for Shoots where the provider ensures the `usr-share-ca-certificates` path
- All existing & new Kubernetes versions 1.17.x  & 1.18.x 
- 1.19.x that are not migrated yet to CSI. 

The existing volume mount `usr-share-ca-certificates` needs to be removed to not have a duplicate `/usr/share/ca-certificates` mount.

**Which issue(s) this PR fixes**:
Fixes #

**Special notes for your reviewer**:

**Release note**:

```action operator
The openstack provider extension is incompatible with Gardener  version > v1.10.0 (if feature flag `MountHostCADirectories` is enabled on the Gardenlet) for Openstack Shoots with certain Kubernetes versions (>= 1.17.x, 1.18.x, 1.19.x without CSI migration complete).  Please consult the compatibility notes under `/docs/compatibility.md`
```

### CODE REVIEW

File: docs/compatibility.md [hunk 1]
```diff
@@ -0,0 +1,8 @@
+## Compatibility
```
Comment (rfranzke) [inline docs/compatibility.md:1]: Does it make sense to put this into the README.md to have it visible all the time?

File: docs/compatibility.md [hunk 2]
```diff
@@ -0,0 +1,8 @@
+## Compatibility 
+
+This file lists known compatibility issues with other Gardener components.
+
+
+| Openstack Extension | Gardener | Action | Notes |
```
Comment (rfranzke) [inline docs/compatibility.md:6]:
```suggestion
| OpenStack Extension | Gardener | Action | Notes |
```

**Review by ialidzhikov — APPROVED** [review]
Apart from the comments above 

/lgtm

**Comment by rfranzke** [comment]
ping @danielfoehrKn

**Review by rfranzke — APPROVED** [review]
/lgtm

**Pull request merged**